### PR TITLE
fix: field_count more than 250

### PR DESCRIPTION
**Pull request by huangnauh**
field_count will be 0xfc (packet content: fc fb 0) when a table has 0xfb columns in mysql

### CODE REVIEW

File: t/lib/sql.lua [hunk 1]
```diff
@@ -0,0 +1,37 @@
+local type = type
```
Comment (agentzh) [inline t/lib/sql.lua]: Better put this Lua code into the test case? It's specific to that test case and won't be usable by other tests, it seems.

File: t/sanity.t [hunk 2]
```diff
@@ -1321,3 +1321,82 @@ success
 --- no_error_log
 [error]
 --- timeout: 20
+
+
```
Comment (agentzh) [inline t/sanity.t]: We use 3 blank lines to separate adjacent test blocks. We actually have a `reindex` tool to auto-format these `.t` files.

File: t/sanity.t [hunk 3]
```diff
@@ -1321,3 +1321,82 @@ success
...
+=== TEST 20: 251 columns
+--- http_config eval: $::HttpConfig
+--- config
+    location /t {
+        content_by_lua_block {
+            local mysql = require "resty.mysql"
+            local sql = require "sql"
```
Comment (agentzh) [inline t/sanity.t]: "sql" is a bad Lua module name since its a variable name usually for SQL query strings.

File: lib/resty/mysql.lua [hunk 4]
```diff
@@ -295,7 +295,7 @@ local function _recv_packet(self)
         typ = "ERR"
     elseif field_count == 0xfe then
         typ = "EOF"
-    elseif field_count <= 250 then
+    else
```
Comment (agentzh) [inline lib/resty/mysql.lua]: I think there's still an upper limit? Will you find it out?

File: t/sanity.t [hunk 5]
```diff
@@ -1321,3 +1321,82 @@ success
...
+                database = "ngx_test",
+                user = "ngx_test",
+                password = "ngx_test"})
+
+            if not ok then
+                ngx.say("failed to connect: ", err, ": ", errno, " ", sqlstate)
+                return
+            end
+
+            local columns = 251
```
Comment (agentzh) [inline t/sanity.t]: What about other numbers like 252 and 253? Will they work too? Why just 251?

Comment (huangnauh) [reply t/sanity.t]: https://dev.mysql.com/doc/refman/5.6/en/column-count-limit.html 
strbyte(data, 1) will be 0xfc when 251 <= columns <= 4096

File: t/sanity.t (hunk 3 above)
Comment (huangnauh) [reply t/sanity.t]: module name change to "create_sql_by_ columns" and add some tests

**Comment by agentzh** [comment]
@huangnauh Your tests fail on my side:

```
t/sanity.t .......... 126/146
#   Failed test 'TEST 22: InnoDB has a limit of 1017 columns per table - response_body - response is expected (repeated req 0, req 0)'
#   at /home/agentzh/git/lua-resty-mysql/../test-nginx/lib/Test/Nginx/Socket.pm line 1500.
#          got: 'bad result: Can't create table `ngx_test`.`test1018` (errno: 185 "Too many columns"): 1005: HY000.
# '
#     expected: 'bad result: Too many columns: 1117: HY000.
# '

#   Failed test 'TEST 22: InnoDB has a limit of 1017 columns per table - response_body - response is expected (repeated req 1, req 0)'
#   at /home/agentzh/git/lua-resty-mysql/../test-nginx/lib/Test/Nginx/Socket.pm line 1500.
#          got: 'bad result: Can't create table `ngx_test`.`test1018` (errno: 185 "Too many columns"): 1005: HY000.
# '
#     expected: 'bad result: Too many columns: 1117: HY000.
# '
# Looks like you planned 146 tests but ran 148.
# Looks like you failed 2 tests of 148 run.
```

**Comment by agentzh** [comment]
@huangnauh Also, the test plan in `t/sanity.t` is not updated to account for your changes:

https://travis-ci.org/openresty/lua-resty-mysql/builds/306133697?utm_source=github_status&utm_medium=notification

The Travis CI build is broken.

Will you look into these issues? Thanks!

**Comment by agentzh** [comment]
@huangnauh Merged. Thanks!

**Pull request closed without merging**